### PR TITLE
Intro: Never change the prune checkbox after the user has touched it

### PR DESCRIPTION
**Pull request by luke-jr**
Re-PR from https://github.com/bitcoin/bitcoin/pull/18729

Now includes a bugfix too (`-prune=2+` disabled the checkbox, but `-prune=0/1` did not; this behaviour is necessary since `-prune` overrides GUI settings)

### CODE REVIEW

**Review by hernanmarino — COMMENTED** [review]
cr  ACK 76cb0898198e19c164f9d649560c472b7bc5fafd and tested ACK

**Review by pablomartin4btc — COMMENTED** [review]
Title and description mismatch of the PR confuse me, I tested bitcoin-qt before and after the change and I don't see the difference in the behaviour (from the UI point of view, haven't checked the proper prunning).
I've ticked & unticked the prune checkbox and set a value, verifying the value is persisted in the config. Also combined the previous starting qt with `-prune` param set to 0, 1 and 600.
Perhaps I've some config that bypass this change? Could you please clarify? Sorry if I miss anything too obvius here.

**Comment by hebasto** [comment]
> `-prune=2+` disabled the checkbox

Looks like a bug in master branch, as it should be like that:
```
$ ./src/bitcoind -prune=2
Error: Prune configured below the minimum of 550 MiB.  Please use a higher number.
```

**Comment by hebasto** [comment]
@ryanofsky Could you be interesting in reviewing of this PR?

**Comment by hebasto** [comment]
Closing due to a long period of inactivity here. Feel free to reopen.

**Comment by achow101** [comment]
Reopening by request

**Comment by luke-jr** [comment]
Rebased

File: src/qt/intro.cpp [hunk 1]
```diff
@@ -287,8 +289,9 @@ void Intro::setStatus(int status, const QString &message, quint64 bytesAvailable
         ui->freeSpace->setText("");
     } else {
         m_bytes_available = bytesAvailable;
-        if (ui->prune->isEnabled() && !(gArgs.IsArgSet("-prune") && gArgs.GetIntArg("-prune", 0) == 0)) {
+        if (ui->prune->isEnabled() && m_prune_checkbox_is_default) {
             ui->prune->setChecked(m_bytes_available < (m_blockchain_size_gb + m_chain_state_size_gb + 10) * GB_BYTES);
+            m_prune_checkbox_is_default = true;
```
[the remark this answers is not in the document]
Comment (luke-jr) [reply src/qt/intro.cpp]: Yes, removed it

**Review by hebasto — APPROVED** [review]
ACK bee0ffbecf4f95b65f4084893924a1ab250ca77c, both commits are improvements of the current behaviour. Tested on Ubuntu 23.10.

**Pull request merged**